soccer/gameplay/plays/testing/four_courners_pass.py

```diff
@@ -0,0 +1,201 @@
+		#max value is the width of the field.
+		variable_square = min(variable_square, width, length)
+
+		#radius of robot
+		self.bot = constants.Robot.Radius
+		# the largest and smallest x and y postion possible for the square
+		self.max_x = (variable_square / 2 - self.bot * 3)
+		self.min_x = -self.max_x
+		self.max_y = length/2 + self.max_x
+		self.min_y = length/2 - self.max_x
+
+		# the four courners
+		self.square_points = [
+			robocup.Point(self.max_x, self.max_y),
+			robocup.Point(self.min_x, self.max_y),
+			robocup.Point(self.min_x, self.min_y),
+			robocup.Point(self.max_x, self.min_y)]
+
+		# speed of the hunting robots
+		self.normal_speed = 2.0
+		# speed of the defending robots can decrease value to make it easier for offense
+		self.defense_speed = variable_square/3.0
+
+		# picks the direction to pass to. TODO make actual smart pass selection
+		self.direction = 1
+
+	# constanly running changes and updates
+	def execute_running(self):
+		# checks if there is an offensive behavior for the various plays and set the normal speed
+		for sub in self.subbehaviors_by_name() :
+			if (sub != 'passer') :
+				robo = self.subbehavior_with_name(sub).robot
+				# checks if there is a chasing robot and set their speed to defense speed
+				if (robo != None) :
+					if (sub == 'chasing'):
+						robo.set_max_speed(self.defense_speed)
+					# otherwise sets to the offensive normal speed rate
+					else:
+						robo.set_max_speed(self.normal_speed)		
+
+		# keep the iso point stuck in the corner
+		if (self.has_subbehavior_with_name('iso')) :
+			self.iso.pos = robocup.Point(-constants.Field.Width/2, 0)
+
+		# chasing robot positon should always follow the ball within a smaller inside box of the
+		# four corners.
+		if (self.has_subbehavior_with_name('chasing')) :
+			self.chaser.pos = self.cut_of_pos(main.ball().pos)
+
+		#draw the four courner field
+		# takes in the square points and form lines and create a square on the field
+		for i in range(len(self.square_points)) :
+			main.system_state().draw_line(robocup.Line(self.square_points[i], 
+			self.square_points[(i + 1) % 4]), (135, 0, 255), "Square")
+
+
+	def on_enter_setup(self):
+		# find where we will consider the ball is closest too and passing from
+		closestPt = min(self.square_points, key=lambda pt: pt.dist_to(main.ball().pos))
+
+		closestPtIdx = self.square_points.index(closestPt)
+
+		#points that the other robots can move to
+		self.otherPts = list(self.square_points)
+		self.otherPts.remove(closestPt)
+
+		# remove the point that's diagonal of the closest point
+		farthestPt = max(self.otherPts,
+						key=lambda pt: pt.dist_to(main.ball().pos))
+
+		self.otherPts.remove(farthestPt)
+
+		# decide which direction you're passing to
+		self.direction = self.safer_pass()
+		# move the other two robots to the other point locations
+		self.add_subbehavior(skills.move.Move(self.square_points[(closestPtIdx + self.direction) % 4]), 'move1', required = False)
```

Comment:
There's only 1 move so you can probably just name it 'move'?

Reply:
There were originally two I'll change it now